core/providers/java/gradle.go

```diff
@@ -0,0 +1,92 @@
+package java
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/railwayapp/railpack/core/generate"
+)
+
+const (
+	DEFAULT_GRADLE_VERSION = "8"
+	GRADLE_CACHE_KEY       = "gradle"
+)
+
+func (p *JavaProvider) usesGradle(ctx *generate.GenerateContext) bool {
+	return ctx.App.HasMatch("gradlew")
+}
+
+func (p *JavaProvider) setGradleVersion(ctx *generate.GenerateContext) error {
+	miseStep := ctx.GetMiseStepBuilder()
+	gradle := miseStep.Default("gradle", DEFAULT_GRADLE_VERSION)
+
+	if envVersion, envName := ctx.Env.GetConfigVariable("GRADLE_VERSION"); envVersion != "" {
+		miseStep.Version(gradle, envVersion, envName)
+	}
+
+	if !ctx.App.HasMatch("gradle/wrapper/gradle-wrapper.properties") {
+		return nil
+	}
+
+	wrapperProps, err := ctx.App.ReadFile("gradle/wrapper/gradle-wrapper.properties")
+	if err != nil {
+		return err
+	}
+
+	versionRegex, err := regexp.Compile(`(distributionUrl[\S].*[gradle])(-)([0-9|\.]*)`)
+	if err != nil {
+		return err
+	}
+
+	if !versionRegex.Match([]byte(wrapperProps)) {
+		return err
+	}
+
+	customVersion := string(versionRegex.FindSubmatch([]byte(wrapperProps))[3])
+
+	parseVersionRegex, err := regexp.Compile(`^(?:[\sa-zA-Z-"']*)(\d*)(?:\.*)(\d*)(?:\.*\d*)(?:["']?)$`)
+	if err != nil {
+		return err
+	}
```

Comment:
I don't think these should error if regex fails. Instead it could add a warning (`ctx.Logger.Warn`) and return `nil`. Erroring will fail the build and we want to avoid that as much as possible. Much better to just use the default gradle version if we can't regex parse the properties file.